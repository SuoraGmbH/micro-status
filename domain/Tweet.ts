export interface Tweet {
  userId: string;
  message: string;
  date: string;
  id: number;
}
