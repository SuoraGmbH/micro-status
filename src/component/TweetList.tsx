import * as React from 'react';
import { Tweet } from '../domain/Tweet';
import { TweetListItem } from './TweetListItem';

interface Props {
  tweets: Tweet[];
}

export class TweetList extends React.Component<Props> {
  public render() {
    const { tweets } = this.props;
    return (
      <ul className="timeline">
        {tweets.map(tweet => (
          <TweetListItem key={tweet.id} tweet={tweet} />
        ))}
      </ul>
    );
  }
}
