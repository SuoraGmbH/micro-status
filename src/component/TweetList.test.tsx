import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import { Tweet } from '../domain/Tweet';
import { TweetList } from './TweetList';

const tweets: Tweet[] = [
  {
    id: 1,
    date: '2018-09-22T12:25:41.801Z',
    message: 'Hello World!',
    userId: 'muhdiekuh',
  },
  {
    id: 2,
    date: '2018-09-23T12:25:41.801Z',
    message: 'Hello from the other side!',
    userId: 'tobmaster',
  },
];

it('renders two items according to snapshot', () => {
  const shallowRenderer = ShallowRenderer.createRenderer();
  const tree = shallowRenderer.render(<TweetList tweets={tweets} />);
  expect(tree).toMatchSnapshot();
});

it('renders one item according to snapshot', () => {
  const shallowRenderer = ShallowRenderer.createRenderer();
  const tree = shallowRenderer.render(<TweetList tweets={[tweets[0]]} />);
  expect(tree).toMatchSnapshot();
});

it('renders zero items according to snapshot', () => {
  const shallowRenderer = ShallowRenderer.createRenderer();
  const tree = shallowRenderer.render(<TweetList tweets={[]} />);
  expect(tree).toMatchSnapshot();
});
