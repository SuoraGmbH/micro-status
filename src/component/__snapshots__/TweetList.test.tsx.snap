// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders one item according to snapshot 1`] = `
<div>
  <TweetListItem
    tweet={
      Object {
        "date": "2018-09-22T12:25:41.801Z",
        "id": 1,
        "message": "Hello World!",
        "userId": "muhdiekuh",
      }
    }
  />
</div>
`;

exports[`renders two items according to snapshot 1`] = `
<div>
  <TweetListItem
    tweet={
      Object {
        "date": "2018-09-22T12:25:41.801Z",
        "id": 1,
        "message": "Hello World!",
        "userId": "muhdiekuh",
      }
    }
  />
  <TweetListItem
    tweet={
      Object {
        "date": "2018-09-23T12:25:41.801Z",
        "id": 2,
        "message": "Hello from the other side!",
        "userId": "tobmaster",
      }
    }
  />
</div>
`;

exports[`renders zero items according to snapshot 1`] = `<div />`;
